File: packages/python/src/endpoints/code_submission.py
"""
Code_Submission API Client
"""

from typing import Dict, List, Optional, Any
import httpx


class Code_SubmissionClient:
    """Client for code_submission endpoints."""

    def __init__(self, base_url: str, client: Optional[httpx.AsyncClient] = None):
        self.base_url = base_url.rstrip('/')
        self.client = client

    async def _request(self, method: str, path: str, **kwargs) -> Any:
        """Make HTTP request."""
        url = self.base_url + path
        if self.client:
            response = await self.client.request(method, url, **kwargs)
            response.raise_for_status()
            return response.json()
        else:
            async with httpx.AsyncClient() as client:
                response = await client.request(method, url, **kwargs)
                response.raise_for_status()
                return response.json()


    async def post_submit_code(self, files: Optional[List[Any]] = None, scan_config: Optional[Any] = None, repository_name: Optional[str] = None, branch: Optional[str] = None, commit_sha: Optional[str] = None):
        """POST /submit/code"""
        url = f"/submit/code"
        params = {}
        data = {", ".join(body_dict)}
        return await self._request("POST", url, params=params, json=data)


    async def post_submit_repository(self, repository_url: Optional[str] = None, snapshot_data: Optional[Any] = None, scan_config: Optional[Any] = None, branch: Optional[str] = None, commit_sha: Optional[str] = None):
        """POST /submit/repository"""
        url = f"/submit/repository"
        params = {}
        data = {", ".join(body_dict)}
        return await self._request("POST", url, params=params, json=data)


    async def post_submit_analysis(self, code_content: Optional[str] = None, language: Optional[str] = None, analysis_type: Optional[str] = None, rules: Optional[List[str]] = None, plugins: Optional[List[str]] = None, context: Optional[Any] = None):
        """POST /submit/analysis"""
        url = f"/submit/analysis"
        params = {}
        data = {", ".join(body_dict)}
        return await self._request("POST", url, params=params, json=data)


    async def get_scans_scan_id_status(self, scan_id: str):
        """GET /scans/{scan_id}/status"""
        url = f"/scans/{scan_id}/status"
        params = {", ".join(query_dict)}
        data = None
        return await self._request("GET", url, params=params, json=data)


    async def get_scans_scan_id_results_summary(self, scan_id: str):
        """GET /scans/{scan_id}/results/summary"""
        url = f"/scans/{scan_id}/results/summary"
        params = {", ".join(query_dict)}
        data = None
        return await self._request("GET", url, params=params, json=data)

File: packages/typescript/src/endpoints/code_submission.ts
/**
 * Code_Submission API Client
 */

import { AxiosInstance } from 'axios'

export class Code_SubmissionClient {
  private httpClient: AxiosInstance

  constructor(private baseURL: string, httpClient: AxiosInstance) {
    this.httpClient = httpClient
  }

  /**
   * POST /submit/code
   */
  async postSubmitcode(files?: UploadFile[], scan_config?: Dict, repository_name?: string, branch?: string, commit_sha?: string): Promise<dict> {
        const params = {}
        const data = {'files: files, scan_config: scan_config, repository_name: repository_name, branch: branch, commit_sha: commit_sha'}
        return this.httpClient.post(`/submit/code`, data, params)
  }
  /**
   * POST /submit/repository
   */
  async postSubmitrepository(repository_url?: string, snapshot_data?: Dict, scan_config?: Dict, branch?: string, commit_sha?: string): Promise<dict> {
        const params = {}
        const data = {'repository_url: repository_url, snapshot_data: snapshot_data, scan_config: scan_config, branch: branch, commit_sha: commit_sha'}
        return this.httpClient.post(`/submit/repository`, data, params)
  }
  /**
   * POST /submit/analysis
   */
  async postSubmitanalysis(code_content?: string, language?: string, analysis_type?: string, rules?: string[], plugins?: string[], context?: Dict): Promise<dict> {
        const params = {}
        const data = {'code_content: code_content, language: language, analysis_type: analysis_type, rules: rules, plugins: plugins, context: context'}
        return this.httpClient.post(`/submit/analysis`, data, params)
  }
  /**
   * GET /scans/{scan_id}/status
   */
  async getScans{scan_id}status(scan_id: string): Promise<any> {
        const params = {'scan_id: scan_id'}
        const data = undefined
        return this.httpClient.get(`/scans/{scan_id}/status`, params)
  }
  /**
   * GET /scans/{scan_id}/results/summary
   */
  async getScans{scan_id}resultssummary(scan_id: string): Promise<any> {
        const params = {'scan_id: scan_id'}
        const data = undefined
        return this.httpClient.get(`/scans/{scan_id}/results/summary`, params)
  }
}
